lib/deck.rb

```diff
@@ -0,0 +1,33 @@
+class Deck 
+  attr_reader :cards
+
+  def initialize(cards) 
+    @cards = cards 
+  end
+
+  def rank_of_card_at(index)
+    @cards[index].rank
+  end
+
+  def high_ranking_cards
+    @cards.select do |card|
+      card.rank >= 11
+    end
+  end
+
+  def percent_high_ranking
+    (high_ranking_cards.length.to_f / @cards.length.to_f * 100).round(2)
+  end
+
+  def remove_card
+    @cards.shift 
+  end
+
+  def add_card(card)
+    @cards.push(card)
+  end
+end
```

Comment:
Another perfect class! Great job using the instance variable `@cards` in your methods and keeping your code DRY by reusing the previously defined `high_ranking_cards` in your `percent_high_ranking` method. Also great choice in using the `select` enumerable in `high_ranking_cards`.